File: templates/problems/02_two_pointers/02_two_sum_ii/two_sum_ii_test.go
package two_sum_ii

import (
	"reflect"
	"testing"
)

func TestTwoSumII(t *testing.T) {
	tests := []struct {
		name     string
		numbers  []int
		target   int
		expected []int
	}{
		{
			name:     "basic case",
			numbers:  []int{2, 7, 11, 15},
			target:   9,
			expected: []int{1, 2},
		},
		{
			name:     "middle elements",
			numbers:  []int{2, 3, 4},
			target:   6,
			expected: []int{1, 3},
		},
		{
			name:     "negative numbers",
			numbers:  []int{-1, 0},
			target:   -1,
			expected: []int{1, 2},
		},
		{
			name:     "larger array",
			numbers:  []int{1, 2, 3, 4, 4, 9, 56, 90},
			target:   8,
			expected: []int{4, 5},
		},
		{
			name:     "first and last",
			numbers:  []int{1, 3, 5, 7},
			target:   8,
			expected: []int{1, 4},
		},
		{
			name:     "two elements",
			numbers:  []int{5, 25},
			target:   30,
			expected: []int{1, 2},
		},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			got := twoSumII(tt.numbers, tt.target)
			if !reflect.DeepEqual(got, tt.expected) {
				t.Errorf("twoSumII(%v, %d) = %v, want %v", tt.numbers, tt.target, got, tt.expected)
			}
		})
	}
}

File: templates/problems/02_two_pointers/02_two_sum_ii/typescript/twoSumII.test.ts
import { describe, it, expect } from 'vitest';
import { twoSumII } from './twoSumII';

describe('TwoSumII', () => {
    it('basic case', () => {
        expect(twoSumII([2, 7, 11, 15], 9)).toEqual([1, 2]);
    });

    it('middle elements', () => {
        expect(twoSumII([2, 3, 4], 6)).toEqual([1, 3]);
    });

    it('negative numbers', () => {
        expect(twoSumII([-1, 0], -1)).toEqual([1, 2]);
    });

    it('larger array', () => {
        expect(twoSumII([1, 2, 3, 4, 4, 9, 56, 90], 8)).toEqual([4, 5]);
    });

    it('first and last', () => {
        expect(twoSumII([1, 3, 5, 7], 8)).toEqual([1, 4]);
    });

    it('two elements', () => {
        expect(twoSumII([5, 25], 30)).toEqual([1, 2]);
    });
});
